authenticator-required-action/src/test/java/de/conciso/keycloak/authentication/required_action/RequiredActionAuthenticatorIT.java

```diff
@@ -0,0 +1,114 @@
+        Arguments.of("peterpommes", "peterpommesPassword", "VERIFY_PROFILE")
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideUserLogins")
+  void testThatUsersHaveCorrectRequiredActionAfterLogin(String userName, String password, String requiredAction) {
+    // Setting the specific required action to be added to the Authentication Config
+    var authenticatorConfig = new AuthenticatorConfigRepresentation();
+    authenticatorConfig.setId(AUTHENTICATOR_CONFIG_ID);
+    authenticatorConfig.setAlias("RequiredAction");
+    var configMap = new HashMap<String, String>();
+    configMap.put("REQUIRED_ACTION", requiredAction);
+    authenticatorConfig.setConfig(configMap);
+
+    var keycloakAdminClient = keycloak.getKeycloakAdminClient().realm(REALM_NAME);
+    keycloakAdminClient.flows().updateAuthenticatorConfig(AUTHENTICATOR_CONFIG_ID, authenticatorConfig);
+    assertThat(keycloakAdminClient.users().searchByUsername(userName, true).get(0).getRequiredActions())
+        .isEmpty();
+
+    executeBrowserFlow(userName, password, requiredAction);
+  }
+
+  void executeBrowserFlow(String userName, String password, String requiredAction) {
+    try (Playwright playwright = Playwright.create()) {
+      BrowserType chromium = playwright.chromium();
+      // comment me in if you want to run in headless mode !
+      //Browser browser = chromium.launch(new BrowserType.LaunchOptions().setHeadless(false));
+      Browser browser = chromium.launch();
+      Page page = browser.newPage();
+      page.navigate(String.format("http://localhost:%s/realms/required-action/account/", keycloak.getHttpPort()));
+
+      page.getByText("Signing in").click();
+      page.waitForURL("**/realms/required-action/protocol/openid-connect/auth**");
+
+      // Login Page
+      page.getByLabel("Username or email").click();
+      page.getByLabel("Username or email").fill(userName);
+
+      page.getByLabel("Password").first().fill(password);
+      page.getByLabel("Password").first().press("Enter");
+
+      // split here, for the specific required action
+      switch (requiredAction) {
+        case "TERMS_AND_CONDITIONS":
+          page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Accept")).click();
+          break;
+        case "UPDATE_PASSWORD":
+          page.getByLabel("New Password").fill("Test123!");
+          page.getByLabel("Confirm Password").fill("Test123!");
+          page.getByLabel("Confirm Password").press("Enter");
+          break;
+        case "VERIFY_PROFILE":
+          assertThat(keycloak.getKeycloakAdminClient().realm(REALM_NAME).users().searchByUsername(userName, true).get(0).getRequiredActions()).containsExactly(requiredAction);
+          break;
+        default:
+          Assertions.fail();
+      }
+
+      page.waitForURL(String.format("http://localhost:%s/realms/required-action/account/#/security/signingin", keycloak.getHttpPort()));
```

Comment:
Can't you get the full base URL from the keycloak (container) instead of just the port?
I think this may fail if you are executing the container on a remote node where it can't be accessed via `localhost`